### ISSUE TITLE
Serialization of GHIssue objects creates very large growing JSON and is very slow

### ISSUE THREAD

**Issue by CodingSpiderFox** [hub4j/github-api#1034]
**Describe the bug**
When I call the API with the below code using an accesstoken and try to write the resulting GHIssue objects, I get a JSON file that grews a few kilobytes every few seconds. I stopped when it reached almost 1 MB after quite some time because I think there is something very wrong. I would not expect such a large result because the Repo only has 2 issues with no comments etc.

**To Reproduce**
```
    List<GHIssue> issues =
      GitHubBuilder().withOAuthToken(accessToken).build().getRepository(fullRepoName).getIssues(GHIssueState.ALL);
    ObjectMapper objectMapper = new ObjectMapper();
    objectMapper.getSerializerProvider().setNullKeySerializer(new DtoNullKeySerializer());
    objectMapper.writeValue(Path.of(destinationPath.toString(), "issues.json").toFile(), issues);
```

DtoNullKeySerializer:

```
package com.codingspiderfox.githubmanager.serialization;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;
import java.io.IOException;

public class DtoNullKeySerializer extends StdSerializer<Object> {
  public DtoNullKeySerializer() {
    this(null);
  }

  public DtoNullKeySerializer(Class<Object> t) {
    super(t);
  }

  @Override
  public void serialize(Object nullKey, JsonGenerator jsonGenerator, SerializerProvider unused)
      throws IOException {
    jsonGenerator.writeFieldName("");
  }
}

```

**Expected behavior**
A full JSON representation in the resulting file with all the data the API returns about the issues (including response header fields)

**Desktop (please complete the following information):**
 - Ubuntu 18.04

**Comment by bitwiseman** [comment]
Looks like some self-referencing cycle is happening.  Your serializer needs to not traverse back.  
Also, what does resulting JSON look like?  You should be able to figure out where the back-referencing is happening.  

Instead of your own ObjectMapper, try using the [`GitHub. getMappingObjectWriter()`](https://github.com/hub4j/github-api/blob/3f9954144a6df515bd7d6c8cdc123536430e893a/src/main/java/org/kohsuke/github/GitHub.java#L1298). 

This hasn't really been a scenario we've looked at with any depth, but I've hard several issues filed for this recently.  Out of curiosity, what are you trying to accomplish?  There's a probably a feature to be aded here to have all the objects in this library have a `toJson()` method.

**Comment by CodingSpiderFox** [comment]
Yes, that was what I was thinking of, too. I started looking through the objects that were fetched in my debugger but it was not so easy because of the many fields.

Update: while I am writing this answer, I opened the file in VSCode and discovered that it's probably because one user follows another and that user follows him back.

So it's one friend of me being followed by one of these two with the bidirectional following relationship.

![image](https://user-images.githubusercontent.com/25173078/107479352-48b67a80-6b7b-11eb-8d1f-866c6d6e44dc.png)
Two usernames keep appearing in the JSON over and over again.

My use case is just backing up my data so, I'd be fine with excluding the followers of mine but the problem will likely arise again deeper into the data structure once the "asignees" are reached for processing because they also refer to users which can follow someone and be followed back by the followed person
